safe/gui/widgets/dock.py

```diff
@@ -745,6 +746,49 @@ def progress_callback(self, current_value, maximum_value, message=None):
         self.progress_bar.setValue(current_value)
         QtGui.QApplication.processEvents()
 
+    def show_print_dialog(self):
+        """Open the print dialog"""
+        if not self.impact_function:
+            # Now try to read the keywords and show them in the dock
+            try:
+                active_layer = self.iface.activeLayer()
+                keywords = self.keyword_io.read_keywords(active_layer)
+
+                provenances = keywords.get('provenance_data', {})
+                if provenances:
+                    self.impact_function = (
+                        ImpactFunction.load_from_output_metadata(keywords))
+
+            except (KeywordNotFoundError,
+                    HashNotFoundError,
+                    InvalidParameterError,
+                    NoKeywordsFoundError,
+                    MetadataReadError,
+                    AttributeError):
+                # Added this check in 3.2 for #1861
+                active_layer = self.iface.activeLayer()
+                if active_layer is None:
+                    if self.conflicting_plugin_detected:
+                        send_static_message(self, conflicting_plugin_message())
+                    else:
+                        send_static_message(self, getting_started_message())
+                else:
+                    show_no_keywords_message(self)
+            except Exception as e:  # pylint: disable=broad-except
+                error_message = get_error_message(e)
+                send_error_message(self, error_message)
+
+        if self.impact_function:
+            dialog = PrintReportDialog(
+                self.impact_function, self.iface, dock=self, parent=self)
+            dialog.show()
+        else:
+            QtGui.QMessageBox.warning(
```

Comment:
safe.utilities.qgis_utilities.display_warning_message_bar thanks less intrusive for UX

https://docs.qgis.org/2.2/en/docs/pyqgis_developer_cookbook/communicating.html#showing-messages-the-qgsmessagebar-class